q_lib/include/q/synth/noise.hpp

```diff
@@ -0,0 +1,99 @@
+/*=============================================================================
+   Copyright (c) 2014-2021 Joel de Guzman. All rights reserved.
+
+   Distributed under the MIT License [ https://opensource.org/licenses/MIT ]
+=============================================================================*/
+#if !defined(CYCFI_Q_NOISE_HPP_AUGUST_3_2021)
+#define CYCFI_Q_NOISE_HPP_AUGUST_3_2021
+
+#include <q/detail/fast_math.hpp>
+#include <q/fx/lowpass.hpp>
+
+namespace cycfi::q
+{
+   ////////////////////////////////////////////////////////////////////////////
+   // white_noise_synth: Synthesizes white noise.
+   // Taken from musicdsp.org
+   ////////////////////////////////////////////////////////////////////////////
+   struct white_noise_synth
+   {
+      constexpr white_noise_synth() {}
+
+      float operator()()
+      {
+         g_x1 ^= g_x2;
+         s = g_x2 * g_fScale;
+         g_x2 += g_x1;
+      }
+
+      int g_x1 = 0x67452301;
+      int g_x2 = 0xefcdab89;
+      float g_fScale = 2.0f / 0xffffffff;
+      float s = 0.0f;
+   };
+
+   auto white_noise = white_noise_synth{};
+
+   ////////////////////////////////////////////////////////////////////////////
+   // pink_noise_synth: Synthesizes pink noise.
+   // Works best for sampling rates between 44.1kHz and 96kHz
+   // Taken from musicdsp.org
+   ////////////////////////////////////////////////////////////////////////////
+   struct pink_noise_synth : white_noise_synth
+   {
+      pink_noise_synth(std::uint32_t sps) : white_noise_synth() {
+         for (unsigned int i = 0; i < 7; ++i) {
+            k[i] = fastexp(-2.0f * M_PI * f[i] / sps);
+         }
+      }
+
+      float operator()()
+      {
+         auto white = white_noise_synth::operator()();
+
+         for (unsigned int i = 0; i < 7; ++i) {
+            b[i] = k[i] * (white + b[i]);
+         }
+
+         pink = 0.05f * (b[0] + b[1] + b[2] + b[3] + b[4] + b[5] + white - b[6]);
+
+         return pink;
+      }
+
+      float f[7] = {
```

Comment:
Make this const_expr

Reply:
Same complaint here too
`error: non-static data member ‘f’ declared ‘constexpr’
       };
`

Reply:
make it static. constexpr has to be static.